src/Workspaces/CSharp/Portable/Extensions/SyntaxEditorExtensions.cs

```diff
@@ -0,0 +1,13 @@
+﻿using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.CSharp.Extensions
+{
+    public static class SyntaxEditorExtensions
+    {
+        public static IDisposableSyntaxEditor RemoveStatementsCleanly(this SyntaxEditor editor)
```

Comment:
SyntaxGenerator is already intended to be the guy that can do things like remove a statement cleanly.  You can already see logic for this sort of thing with methods like 'Isolate' and 'RemoveNodeInternal'  These are already trying to do sensible operations around removal so that someone can just ask to remove a node, but have that result in something positive without having to think about it too much.

For example, if you have `[A]class C { }` and you remove the `Attribute` node for `A`, then SyntaxGenerator will actually remove the entire `[A]`.  It will also try to sensibly figure out where trivia should go.

As mentioned before, you can either:
1. just fix up this specific code-fix-provider to just do a better job with trivia.
2. fix up syntax generator to do a better job with removing statements.

Note: my preference is '1' as it's localized and has hte least chance of breaking someone else unintentionally.  However, i'm amenable to '2' if we can be care about it.